apps/shared/avifjpeg.c

```diff
@@ -225,6 +226,26 @@ static avifBool avifJPEGReadCopy(avifImage * avif, struct jpeg_decompress_struct
     return AVIF_FALSE;
 }
 
+// Reads 4-byte unsigned integer in big-endian format from the raw bitstream src.
+static uint32_t avifJPEGReadUint32BigEndian(const uint8_t * src)
+{
+    return ((uint32_t)src[0] << 24) | ((uint32_t)src[1] << 16) | ((uint32_t)src[2] << 8) | ((uint32_t)src[3] << 0);
+}
+
+// Returns the pointer in str to the first occurrence of substr. Returns NULL if substr cannot be found in str.
+static const uint8_t * avifJPEGFindSubstr(const uint8_t * str, size_t strLength, const uint8_t * substr, size_t substrLength)
+{
+    for (size_t index = 0; index + substrLength <= strLength; ++index) {
+        if (!memcmp(&str[index], substr, substrLength)) {
+            return &str[index];
+        }
+    }
+    return NULL;
+}
+
+#define AVIF_EXTENDED_XMP_GUID_LENGTH 32
+#define AVIF_XMP_NOTE_TAG_LENGTH 24
```

Comment:
Nit/optional: Would be good to note that this is the length of `"xmpNote:HasExtendedXMP=\""`

Reply:
I moved the string next to the size to make it clear. That was my original intent: keep the string, the size and where it is used close.